### DISCUSSION TITLE
Apitally - Simple, privacy-focused API monitoring & analytics for Fastify

### DISCUSSION THREAD

**Discussion by itssimon** [fastify/fastify#5532] [Show and tell]
Hey Fastify community,

I'd like to introduce you to my indie product [Apitally](https://apitally.io/fastify), a simple, privacy-focused API monitoring and analytics tool for Fastify. What sets it apart is that it's super easy to set up and gives you opinionated, useful dashboards out of the box.

### Feature highlights

- API traffic, error, and response time metrics per endpoint
- Tracking of individual API consumers
- Request logs with correlated application logs and traces
- Uptime monitoring, CPU & memory usage
- Custom alerts via email, Slack, or Microsoft Teams

✨ [New CLI & skill for agents](https://apitally.io/blog/apitally-cli-and-skill-for-agents): Run arbitrary SQL queries against API metrics and request logs in agent-driven investigations.

### Integration

The open-source [Apitally SDK](https://github.com/apitally/apitally-js) integrates with Fastify via a lightweight plugin, and syncs data in the background at regular intervals. It minimizes data collection by default, and gives you full control over what's captured.

To set it up for your apps, you just need to add a few lines of code. No infrastructure changes required, no dashboards to build.

```javascript
import Fastify from "fastify";
import { apitallyPlugin } from "apitally/fastify";

const fastify = Fastify({ logger: true });

await fastify.register(apitallyPlugin, {
  clientId: "your-client-id",
  env: "dev", // or "prod" etc.

  // Optional: configure request logging
  requestLogging: {
    enabled: true,
    logRequestHeaders: true,
    logRequestBody: true,
    logResponseBody: true,
    captureLogs: true,
  },
});
```

Here's the full [setup guide for Fastify](https://docs.apitally.io/setup-guides/fastify) if you'd like to try it out.

### Dashboard

<p>
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://assets.apitally.io/screenshots/overview-dark.png">
    <source media="(prefers-color-scheme: light)" srcset="https://assets.apitally.io/screenshots/overview-light.png">
    <img alt="Apitally dashboard" src="https://assets.apitally.io/screenshots/overview-light.png">
  </picture>
</p>

### Motivation

I built Apitally because I was frustrated with enterprise monitoring tools like Datadog, which were too complex for what I needed: a clear view of how my APIs were doing. I've been working on it for over 2 years and it's now used by 100+ engineering teams. I hope that you guys find it useful too.

**Comment by GreenFlag31** [comment]
Look nice, but there is no way I would send (potentially) sensitive data over a distant and private server (I think this would even violate the GDPR principles in Europe). You should provide a way to keep it local.

**Reply by itssimon** [reply]
Thanks! Definitely planning to offer self-hosting in the future.

That said, the Apitally SDK gives you fine-grained control over what's included in logs, and you can easily mask sensitive data. We have users in the healthcare industry with strict data privacy requirements, and also many in the EU.

Here's a reference with all the available parameters and default masking rules:
https://docs.apitally.io/reference/javascript
